tests/test_signal_handlers.py

```diff
@@ -0,0 +1,344 @@
+        "wagtail_localize_smartling.signal_handlers.LandedTranslationTask.objects.filter",
+        return_value=[mock_task],
+    )
+    mock_instance = mocker.Mock()
+    mock_instance.pk = 1234
+    mock_content_type = mocker.patch(
+        "wagtail_localize_smartling.signal_handlers.ContentType.objects.get_for_model",
+        return_value=mocker.Mock(),
+    )
+
+    _close_translation_landed_task(mock_instance)
+
+    mock_content_type.assert_called_once_with(mock_instance)
+    mock_tasks_filter.assert_called_once_with(
+        content_type=mock_content_type.return_value,
+        object_id=1234,
+    )
+    mock_task.complete.assert_called_once()
+
+
+def test__close_translation_landed_task__no_tasks_found(mocker):
+    mock_tasks_filter = mocker.patch(
+        "wagtail_localize_smartling.signal_handlers.LandedTranslationTask.objects.filter",
+        return_value=[],
+    )
+    mock_instance = mocker.Mock()
+    mock_instance.pk = 5432
+    mock_content_type = mocker.patch(
+        "wagtail_localize_smartling.signal_handlers.ContentType.objects.get_for_model",
+        return_value=mocker.Mock(),
+    )
+
+    _close_translation_landed_task(mock_instance)
+
+    mock_content_type.assert_called_once_with(mock_instance)
+    mock_tasks_filter.assert_called_once_with(
+        content_type=mock_content_type.return_value,
+        object_id=5432,
+    )
+
+
+@pytest.mark.parametrize("fake_a_page", [True, False])
+def test_close_translation_landed_task_on_page_published(fake_a_page, mocker):
+    if fake_a_page:
+        instance = Page()
+    else:
+        instance = User()
+
+    mock_close_translation_landed_task = mocker.patch(
+        "wagtail_localize_smartling.signal_handlers._close_translation_landed_task"
+    )
+
+    close_translation_landed_task_on_page_published(
+        sender=Page,
+        instance=instance,
+    )
+
+    if fake_a_page:
+        mock_close_translation_landed_task.assert_called_once_with(instance)
+    else:
+        assert not mock_close_translation_landed_task.called
+
+
+@pytest.mark.parametrize("is_a_snippet", [True, False])
+def test_close_translation_landed_task_on_snippet_published(is_a_snippet, mocker):
+    mock_instance = mocker.Mock()
+    mock_close_translation_landed_task = mocker.patch(
+        "wagtail_localize_smartling.signal_handlers._close_translation_landed_task"
+    )
+    mock_model_is_registered_as_snippet = mocker.patch(
+        "wagtail_localize_smartling.signal_handlers._model_is_registered_as_snippet",
+        return_value=is_a_snippet,
+    )
+
+    close_translation_landed_task_on_snippet_published(
+        sender=mock_instance.__class__,
+        instance=mock_instance,
+    )
+
+    mock_model_is_registered_as_snippet.assert_called_once_with(mock_instance)
+    if is_a_snippet:
+        mock_close_translation_landed_task.assert_called_once_with(mock_instance)
+    else:
+        assert not mock_close_translation_landed_task.called
```

Comment:
There's also `mock_close_translation_landed_task.assert_not_called()` if that feels better